packages/eslint-plugin/docs/rules/switch-exhaustiveness-check.md

```diff
@@ -101,6 +103,25 @@ switch (day) {
 }
 ```
 
+## Options
+
+### `requireDefaultForNonUnion`
+
+Examples of additional **incorrect** code for this rule with `{ requireDefaultForNonUnion: true }`:
+
+```ts option='{ "requireDefaultForNonUnion": true }' showPlaygroundButton
+const value: number = Math.floor(Math.random() * 3);
+
+switch (value) {
+  case 0:
+    return 0;
+  case 1:
+    return 1;
+}
+```
+
+Since `value` is a non-union type it requires the switch case to have a default clause.
```

Comment:
```suggestion
Since `value` is a non-union type it requires the switch case to have a default clause only with `requireDefaultForNonUnion` enabled.
```